hazelcast/src/test/java/com/hazelcast/spi/impl/SlowOperationDetectorTest.java

```diff
@@ -0,0 +1,483 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.instance.GroupProperties;
+import com.hazelcast.instance.TestUtil;
+import com.hazelcast.management.TimedMemberStateFactory;
+import com.hazelcast.map.EntryBackupProcessor;
+import com.hazelcast.map.EntryProcessor;
+import com.hazelcast.monitor.TimedMemberState;
+import com.hazelcast.spi.AbstractOperation;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.SlowTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static com.hazelcast.instance.TestUtil.getHazelcastInstanceImpl;
+import static java.lang.String.format;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(SlowTest.class)
+public class SlowOperationDetectorTest extends HazelcastTestSupport {
+
+    private static final String DEFAULT_KEY = "key";
+    private static final String DEFAULT_VALUE = "value";
+
+    @Test
+    public void testSlowEntryProcessor() throws InterruptedException {
+        HazelcastInstance instance = getSingleNodeCluster(3000);
+        IMap<String, String> map = getMapWithSingleElement(instance);
+
+        for (int i = 0; i < 3; i++) {
+            map.executeOnEntries(new SlowEntryProcessor(4));
+        }
+        map.executeOnEntries(new SlowEntryProcessor(7));
+
+        shutdownOperationService(instance);
+
+        Collection<SlowOperationLog> logs = getSlowOperationLogs(instance);
+        assertEqualsStringFormat("Expected %d slow operation logs, but was %d", 1, logs.size());
+
+        SlowOperationLog firstLog = logs.iterator().next();
+        assertTotalInvocations(firstLog, 4);
+        assertEntryProcessorOperation(firstLog);
+        assertStackTraceContainsClassName(firstLog, "$SlowEntryProcessor");
+
+        Collection<SlowOperationLog.Invocation> invocations = firstLog.getInvocations();
+        assertEqualsStringFormat("Expected %d invocations, but was %d", 4, invocations.size());
+        for (SlowOperationLog.Invocation invocation : invocations) {
+            assertDurationBetween(invocation.getDuration(), 3000, 7500);
+        }
+    }
+
+    @Test
+    public void testMultipleSlowEntryProcessorClasses() throws InterruptedException {
```

Comment:
You are testing 2 things; one is the json stuff the other is the scanning behavior. I would not do that because it makes your test difficult to understand.